### PR TITLE
doc(storage): describe connection pool

### LINKED ISSUE #1554: Document the connection pooling model in Cloud Storage library.
We should give guidance to our users about how to get more connections, how are they used, etc. This is important for them to know how to get the most out of the library.

### PR DESCRIPTION
**Pull request by coryan**
Fixes #1554

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7637)

### CODE REVIEW

File: google/cloud/storage/client.h [hunk 1]
```diff
@@ -68,6 +68,32 @@ struct ClientImplDetails;
  * For example, access tokens need to be refreshed from time to time, and this
  * may impact the performance of some operations.
  *
+ * @par Connection Pool
+ * By default this class uses HTTPS to communicate with the service. Creating a
+ * new HTTPS session is relatively expensive, as it must go through the TCP/IP
+ * and SSL handshakes. To minimize this overhead the class maintains a
+ * connection pool to the service. After each request completes the connection
+ * is returned to the pool, and reuse in future requests. Note that for
```
Comment (dbolduc) [inline google/cloud/storage/client.h:76]: s/reuse/reused/

File: google/cloud/storage/client.h [hunk 2]
```diff
@@ -68,6 +68,32 @@ struct ClientImplDetails;
...
+ * downloads (implemented by the ReadObject() member function) the connection
+ * remains in use until the download completes. Therefore, having multiple
+ * downloads open at the same time requires multiple connections.
+ *
+ * The application can limit the maximum size of this connection pool using
+ * `storage::ConnectionPoolSizeOption`. If returning a connection to the pool
+ * would make the pool larger than this limit then the oldest connection in the
```
Comment (dbolduc) [inline google/cloud/storage/client.h:83]: nit: Would "oldest inactive connection" be more accurate?

The paragraph below suggests that an active download will not be closed.

Your call on whether to update the text or not.

**Review by dbolduc — APPROVED** [review]

File: google/cloud/storage/client.h (hunk 2 above)
Comment (coryan) [reply google/cloud/storage/client.h:83]: All connections in the pool are inactive ... and yes, active downloads are not closed (not while in progress).

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7637)* status: 0 of 1 files reviewed, 3 unresolved discussions (waiting on @coryan, @dbolduc, and @scotthart)

___
*[google/cloud/storage/client.h, line 84 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7637#-MotkxUREl69sT8-INGS:-MotkxUREl69sT8-INGT:b-iybe5m) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/35b9db0c0ac210b5e0a274488072097ff8bfb1e4/google/cloud/storage/client.h#L84)):*
> ```
> inactivate
> ```
inactive

File: google/cloud/storage/client.h (hunk 2 above)
Comment (dbolduc) [reply google/cloud/storage/client.h:83]: hmm ok.

s/connecitons in the pool are inactivate/connections in the pool are inactive/

File: google/cloud/storage/client.h (hunk 1 above)
Comment (coryan) [reply google/cloud/storage/client.h]: Fixed.

File: google/cloud/storage/client.h (hunk 2 above)
Comment (coryan) [reply google/cloud/storage/client.h:83]: Fixed.

**Review by coryan — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7637)* status: 0 of 1 files reviewed, 2 unresolved discussions (waiting on @dbolduc and @scotthart)

___
*[google/cloud/storage/client.h, line 84 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7637#-MotkxUREl69sT8-INGS:-Mou3BgP9r66lr2ZEq40:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/35b9db0c0ac210b5e0a274488072097ff8bfb1e4/google/cloud/storage/client.h#L84)):*
<details><summary><i>Previously, scotthart (Scott Hart) wrote…</i></summary><blockquote>

> ```
> inactivate
> ```
inactive
</blockquote></details>

Done.

**Pull request merged**